src/tests/rules.rs

```diff
@@ -61,3 +61,71 @@ fn no_more_than_5_uses_of_foo_rule() -> Arc<dyn Rule> {
         ]
     }
 }
+
+#[test]
+fn test_rule_options() {
+    RuleTester::run(
+        no_more_than_n_uses_of_foo_rule(),
+        rule_tests! {
+            valid => [
+                {
+                    code => r#"
+                        fn foo() {
+                            let foo = foo;
+                            foo();
+                            foo();
+                        }
+                    "#,
+                    options => 5,
+                }
+            ],
+            invalid => [
+                {
+                    code => r#"
+                        fn foo() {
+                            let foo = foo;
+                            foo();
+                            foo();
+                            foo();
+                        }
+                    "#,
+                    options => 5,
+                    errors => [r#"Can't use 'foo' more than 5 times"#],
+                },
+            ]
+        },
+    );
+}
+
+fn no_more_than_n_uses_of_foo_rule() -> Arc<dyn Rule> {
+    // parse_options => |options| -> usize {
+    //     if options.len() != 1 {
+    //         return Err("Expected single option for # of uses");
+    //     }
+    // }
+    rule! {
+        name => "no_more_than_n_uses_of_foo",
+        options_type => usize,
```

Comment:
So ya mental model (for this for a rule author using the `rule!` macro) is:

If your rule takes configuration options, you should probably almost always declare their "strong type" via `options_type` here

(if you don't then the type you'll be working with in trying to pluck things off of `options` into your local `state` will be `serde_json::Value`)

And then ya you grab values from `options` into `per-run` state fields

Where their initializers have scope visibility of the name `options` (per above either strongly-typed or `serde_json::Value`)

If it seems "precious" to one to "pluck individual fields" off of `options` then I'd say just do a single
```
state => {
    [per-run]
    options: WhateverTheOptionsTypeWas = options,
}
```
but for whatever reason I'm not as inclined to "magically wire that up"

Probably because I like the idea of them being "top-level" fields on `self`?

But I could see it being simpler to just magically wire up `self.options` and then you can access `self.options.whatever`?